README.md

```diff
@@ -52,10 +52,376 @@ Run the Cypress E2E test suite, either in the Cypress GUI or in the terminal.
+
+The `useRtkQuery` and `useRtkMutation` hooks are similar to those from `@dhis2/app-service-data`, with a few key differences:
+
+-   The hooks from `@dhis2/app-service-data` accept a second positional `options` object, while the RTK Query hooks do not. However, for each options field, there is usually a more ergonomic alternative in RTK Query:
+    -   Since `useRtkQuery` and `useRtkMutation` accept dynamically constructed query/mutation objects, the `variables` field is redundant (as are the callback forms of the `id` and `params` fields).
+    -   To perform a lazy query, use the `useRtkLazyQuery` hook.
+    -   Instead of `onSuccess`/`onError` callbacks, use `useEffect` to monitor state transitions, or use the `trigger().unwrap()` function returned from `useRtkLazyQuery` for promise-based handling.
+-   The `useDataQuery` hook only accepts a nested query definition (e.g., `{ me: { resource: 'me' } }`), but `useRtkQuery` also accepts a simple query object (e.g., `{ resource: 'me' }`). This makes accessing response data more straightforward (e.g., `data.name` instead of `data.me.name`).
+-   The data returned from the hook is slightly different; see the [Redux Toolkit Hooks docs](https://redux-toolkit.js.org/rtk-query/api/created-api/hooks) for details.
+
+#### Custom Endpoints
+
+While the generic hooks are suitable for most use cases, you can also create custom endpoints with auto-generated hooks by adding endpoints in `/src/api/api.ts` or using the `injectEndpoints` or `enhanceEndpoints` functions as [documented here](https://redux-toolkit.js.org/rtk-query/api/created-api/code-splitting). Each custom endpoint can declare its own `queryFn`, where it can access the `engine`, metadata store, and app cached data via the `api.extra` argument. Custom endpoints can also specify their own data types, so consuming components do not have to declare them.
+
+**When to create a custom endpoint:**
+
+-   Analytics requests, where you want to create an analytics request instance in the function body.
+-   Chained requests, where the response from one request is needed to initiate another.
+-   Requests to resources outside the DHIS2 Web API scope (not on `${baseUrl}/api/${version}`).
+-   Common requests that occur in multiple places in the codebase, to avoid repetition and prevent duplicate requests.
+-   Paginated requests (infinite queries), which may benefit from a dedicated endpoint.
+-   Requests to the [Gist API](https://docs.dhis2.org/en/develop/using-the-api/dhis-core-version-master/metadata-gist.html).
+
+#### Using `useRtkQuery`
+
+<details>
+<summary>Show code</summary>
+
+```typescript
+import type { MeDto, PickWithFieldFilters } from '@types'
+import React from 'react'
+import { useRtkQuery } from '../../hooks'
+
+const fieldsFilter = ['id', 'name', 'email', 'settings'] as const
```

Comment:
Is it necessary to use this const assertion?

Reply:
As discussed this is a requirement by the `PickWithFieldsFilter` generic type